tests/kubesonic/kubesonic_test.md

```diff
@@ -0,0 +1,105 @@
+
+## Scope
+
+### Testbed
+
+The test is able to run on KVM testbed and DUT testbed
+
+### K8s version
+
+This case only runs on k8s v1.22.2
+
+### Supported hwsku
+
+7060 / 7050qx is not supported
+
+## Topology
+
+Supported any topology
+
+## Test DUT join and disjoin from k8s cluster
+
+### Setup
+
+Use Minikube to startup a single master k8s cluster on the server of the testbed where the ptf container and simulating neighbor contianers are running.
+
+#### Install Minikube and start Minikube
+1. Install Minikube and start Minikube on the server of the testbed by following the instructions in the [Minikube installation guide](https://minikube.sigs.k8s.io/docs/start/).
+
+#### Update the server kernel parameter
+2. Update the server kernel parameter by running the following command ```sysctl fs.protected_regular=0```. [Check the issue here](https://github.com/kubernetes/minikube/issues/7053)
+
+#### Update the kubelet configmap
+3. The Minikube cluster's pki cert root directory is `/var/lib/minikube/certs`, DUT's kubelet is using `/etc/kubernetes/pki` as the pki cert root directory. So need to update the kubelet configmap.
+
+#### Deploy the daemonset
+4. Deploy the daemonset with nodeSelector `deployDaemonset=true` so that we can trigger the daemonset pod to be deployed or to be removed on the DUT with label and unlabel.
+
+#### Prepare the cert
+5. DUT will need a cert to join the k8s cluster, so need to prepare the cert for the DUT. Copy the cert from the Minikube cluster to the DUT.
+
+#### Prepare Minikube VIP DNS
+6. Minikube cluster's VIP is ```control-plane.minikube.internal```, DUT will not resolve this VIP by default, so need to add the VIP to the DUT's `/etc/hosts` file.
+
+### Trigger DUT to join
+
+1. ```sudo config kube server ip <server_ip> && sudo config kube server disable off``` to trigger the DUT to join the k8s cluster, check if the DUT is in the k8s cluster by running `kubectl get nodes` on the server.
+
+### Trigger DUT to be deployed daemonset pod
+2. ```kubectl label node <node_name> deployDaemonset=true``` to trigger the k8s daemonset pod to be deployed on the DUT, check if the pod is running on the DUT by running `kubectl get pods` on the server.
+
+### Trigger daemonset pod to be removed from the DUT
+3. ```kubectl label node <node_name> deployDaemonset-``` to trigger the k8s daemonset pod to be removed from the DUT, check if the pod is removed from the DUT by running `kubectl get pods` on the server.
+
+### Trigger DUT to disjoin
+4. ```sudo config kube server disable on``` to trigger the DUT to disjoin the k8s cluster, check if the DUT is not in the k8s cluster by running `kubectl get nodes` on the server.
+
+### Teardown
+
+#### Clean up
+Remove the Minikube cluster and restore all changes on the server of the testbed and the DUT.
+
+#### Keep the Minikube cluster running
+If the server of testbed is shared with other tests, need keep the Minikube cluster running to avoid conflict.
+
+### Server lockup
```

Comment:
server lockup may be confusing, here we use a file lock to resolve test case or env conflict

Reply:
fixed